mlflow/keras.py

```diff
@@ -311,3 +316,64 @@ def load_model(model_uri, **kwargs):
         local_model_path,
         flavor_conf.get("data", _MODEL_SAVE_PATH))
     return _load_model(model_path=keras_model_artifacts_path, keras_module=keras_module, **kwargs)
+
+
+class __MLflowKerasCallback(Callback):
+    """
+    Callback for auto-logging metrics and parameters.
+    Records available logs after each epoch.
+    Records model structural information as params after training finishes.
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def on_epoch_end(self, epoch, logs=None):
+        if not logs:
+            return
+        try:
+            mlflow.log_metrics(logs, step=epoch)
+        except mlflow.exceptions.MlflowException as e:
+            warnings.warn("Logging to MLflow failed: " + str(e))
+
+    def on_train_end(self, logs=None):
+        try:
+            mlflow.log_param('num_layers', len(self.model.layers))
+            mlflow.log_param('optimizer_name', type(self.model.optimizer).__name__)
+            if hasattr(self.model.optimizer, 'lr'):
+                mlflow.log_param('learning_rate', self.model.optimizer.lr)
+            if hasattr(self.model.optimizer, 'epsilon'):
+                mlflow.log_param('epsilon', self.model.optimizer.epsilon)
+            sum_list = []
+            self.model.summary(print_fn=sum_list.append)
+            summary = '\n'.join(sum_list)
+            mlflow.set_tag('summary', summary)
+            log_model(self.model, artifact_path='model')
+        except mlflow.exceptions.MlflowException as e:
+            warnings.warn("Logging to Mlflow failed: " + str(e))
+
+
+def autolog():
```

Comment:
Add experimental annotation?

Reply:
Would also be nice to flesh out this docstring a bit more (e.g. the TF docstring has):
```
    """
    Enable automatic logging from TensorFlow to MLflow. If applicable,
    model checkpoints are logged as artifacts to a 'models' directory, along
    with any TensorBoard log data.

    Refer to the tracking documentation for
    information on what is logged with different TensorFlow workflows.

    :param metrics_every_n_steps: The frequency with which metrics should be logged.
                                  Defaults to 100. Ex: a value of 100 will log metrics
                                  at step 0, 100, 200, etc.

    """
```

Would be good to replicate that, maybe we can actually include the URL of the tracking docs in^ (e.g. https://mlflow.org/docs/latest/tracking.html)